src/analysis/normed_space/lattice_ordered_group.lean

```diff
@@ -0,0 +1,181 @@
+  rw ← neg_inf_eq_sup_neg,
+  finish,
+end
+
+/--
+Let `α` be a normed lattice ordered group, then the order dual is also a
+normed lattice ordered group.
+-/
+@[priority 100] -- see Note [lower instance priority]
+instance {α : Type*} [h: normed_lattice_add_comm_group α] :
+  normed_lattice_add_comm_group (order_dual α) :=
+  { add_le_add_left := begin
+    intros a b h₁ c,
+    rw ← order_dual.dual_le,
+    rw ← order_dual.dual_le at h₁,
+    apply h.add_le_add_left,
+    exact h₁,
+  end,
+  solid := begin
+    intros a b h₂,
+    apply dual_solid,
+    rw ← order_dual.dual_le at h₂,
+    finish,
+  end, }
+
+/--
+Let `α` be a normed lattice ordered group, let `a` be an element of `α` and let `|a|` be the
+absolute value of `a`. Then `∥|a|∥ = ∥a∥`.
+-/
+lemma norm_abs_eq_norm {α : Type*} [normed_lattice_add_comm_group α] (a : α) : ∥|a|∥ = ∥a∥ :=
+begin
+  rw le_antisymm_iff,
+  split,
+  { apply normed_lattice_add_comm_group.solid,
+    rw ← lattice_ordered_comm_group.abs_idempotent a, },
+  { apply normed_lattice_add_comm_group.solid,
+    rw ← lattice_ordered_comm_group.abs_idempotent a, }
+end
+
+/--
+Let `α` be a normed lattice ordered group. Then the infimum is jointly continuous.
+-/
+@[priority 100] -- see Note [lower instance priority]
+instance normed_lattice_add_comm_group_has_continuous_inf {α : Type*}
+  [normed_lattice_add_comm_group α] : has_continuous_inf α :=
+⟨ continuous_iff_continuous_at.2 $ λ q, tendsto_iff_norm_tendsto_zero.2 $
+begin
+  have : ∀ p : α × α, ∥p.1 ⊓ p.2 - q.1 ⊓ q.2∥ ≤ ∥p.1 - q.1∥ + ∥p.2 - q.2∥,
+  {
+    intros,
+    nth_rewrite_rhs 0  ← norm_abs_eq_norm,
+    nth_rewrite_rhs 1  ← norm_abs_eq_norm,
+    apply le_trans _ (norm_add_le (|p.fst - q.fst|) (|p.snd - q.snd|)),
+    apply normed_lattice_add_comm_group.solid,
+    rw lattice_ordered_comm_group.abs_pos_eq (|p.fst - q.fst| + |p.snd - q.snd|),
+    { calc |p.fst ⊓ p.snd - q.fst ⊓ q.snd| =
+        |p.fst ⊓ p.snd - q.fst ⊓ p.snd + (q.fst ⊓ p.snd - q.fst ⊓ q.snd)| :
+          by { rw sub_add_sub_cancel, }
+        ... ≤ |p.fst ⊓ p.snd - q.fst ⊓ p.snd| + |q.fst ⊓ p.snd - q.fst ⊓ q.snd| :
+          by {apply lattice_ordered_comm_group.abs_triangle,}
+        ... ≤ |p.fst - q.fst | + |p.snd - q.snd| : by {
+          apply add_le_add,
+          apply
+            (sup_le_iff.elim_left (lattice_ordered_comm_group.Birkhoff_inequalities _ _ _)).right,
+          -- Should I be surprised that this isn't infered automatically?
+          apply normed_lattice_add_comm_group_to_covariant_class,
+          rw inf_comm,
+          nth_rewrite 1 inf_comm,
+          apply
+            (sup_le_iff.elim_left (lattice_ordered_comm_group.Birkhoff_inequalities _ _ _)).right,
+          -- Should I be surprised that this isn't infered automatically?
+          apply normed_lattice_add_comm_group_to_covariant_class,
+        }, },
+    { apply add_nonneg,
+      apply lattice_ordered_comm_group.abs_pos,
+      apply lattice_ordered_comm_group.abs_pos, }
```

Comment:
You could write this as a single `exact` if you like. Not a big deal either way.

Reply:
Thanks! Done.